### PR TITLE
feat(codewhisperer): add client component latency

### PR DESCRIPTION
**Pull request by leigaol**
## Problem
Add a new client component latency definition for CWSPR
## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: telemetry/definitions/commonDefinitions.json [hunk 1]
```diff
@@ -1963,6 +1993,49 @@
                 }
             ]
         },
+        {
+            "name": "codewhisperer_clientComponentLatency",
+            "description": "The duration from user last modification to the first recommendation shown in milliseconds",
```
Comment (zixlin7) [inline telemetry/definitions/commonDefinitions.json:1998]: do we need to update this description?

Comment (leigaol) [reply telemetry/definitions/commonDefinitions.json]: Update. Thanks for the catch

**Review by andrewyuq — APPROVED** [review]

File: telemetry/definitions/commonDefinitions.json [hunk 2]
```diff
@@ -550,6 +550,36 @@
             "description": "The type of the user trigger to send request to CodeWhisperer service",
             "allowedValues": ["OnDemand", "AutoTrigger"]
         },
+        {
+            "name": "codewhispererFirstCompletionLatency",
+            "type": "double",
+            "description": "The latency from user performs the triggering CW action to plugin call GenerateCompletions API for the first recommendation"
```
Comment (andrewyuq) [inline telemetry/definitions/commonDefinitions.json:556]: The time it takes for the response to be received after the plugin makes a first GenerateCompletions API call.

File: telemetry/definitions/commonDefinitions.json [hunk 3]
```diff
@@ -550,6 +550,36 @@
...
+        },
+        {
+            "name": "codewhispererEndToEndLatency",
+            "type": "double",
+            "description": "The latency from plugin start invocation to plugin call GenerateCompletions API to first completion shown in the IDE"
```
Comment (andrewyuq) [inline telemetry/definitions/commonDefinitions.json:561]: The time it takes for the first completion to be shown in the IDE after the user performs the CW trigger action.

File: telemetry/definitions/commonDefinitions.json [hunk 4]
```diff
@@ -550,6 +550,36 @@
...
+        },
+        {
+            "name": "codewhispererAllCompletionsLatency",
+            "type": "double",
+            "description": "The latency from plugin makes a first call to GenerateCompletions API to receiving the response for the last call to the GenerateCompletions API."
```
Comment (andrewyuq) [inline telemetry/definitions/commonDefinitions.json:566]: The time it takes for the last GenerateCompletions response to be received after plugin makes a first call to GenerateCompletions API

File: telemetry/definitions/commonDefinitions.json [hunk 5]
```diff
@@ -550,6 +550,36 @@
...
+        },
+        {
+            "name": "codewhispererPostprocessingLatency",
+            "type": "double",
+            "description": "The latency from plugin gets Completions object to completion shown in the IDE for the first completion."
```
Comment (andrewyuq) [inline telemetry/definitions/commonDefinitions.json:571]: The time it takes for the first completions to be displayed in the IDE after the plugin receives the initial `Completions` object.

File: telemetry/definitions/commonDefinitions.json [hunk 6]
```diff
@@ -550,6 +550,36 @@
...
+        },
+        {
+            "name": "codewhispererPreprocessingLatency",
+            "type": "double",
+            "description": "The latency from user performs the triggering CodeWhisperer action to plugin call GenerateCompletions API for the first recommendation"
```
Comment (andrewyuq) [inline telemetry/definitions/commonDefinitions.json:576]: The time it takes for the plugin to make the first GenerateCompletions API call after the user performs the CW trigger action.

File: telemetry/definitions/commonDefinitions.json [hunk 7]
```diff
@@ -550,6 +550,36 @@
...
+        },
+        {
+            "name": "codewhispererCredentialFetchingLatency",
+            "type": "double",
+            "description": "The latency to get the SSO credential within the invocation."
```
Comment (andrewyuq) [inline telemetry/definitions/commonDefinitions.json:581]: The time it takes to get the Sono/SSO credential for the invocation.

**Review by andrewyuq — APPROVED** [review]

**Review by JadenSimon — APPROVED** [review]

**Review by JadenSimon — APPROVED** [review]

**Pull request merged**